pkg/reconciler/pipelinerun/timeout.go

```diff
@@ -81,20 +81,29 @@ func init() {
 	}
 }
 
+// setPipelineRunTimeoutCondition sets the status of the PipelineRun to timed out.
+func setPipelineRunTimeoutCondition(ctx context.Context, pr *v1beta1.PipelineRun) {
+	pr.Status.SetCondition(&apis.Condition{
+		Type:    apis.ConditionSucceeded,
+		Status:  corev1.ConditionFalse,
+		Reason:  v1beta1.PipelineRunReasonTimedOut.String(),
+		Message: fmt.Sprintf("PipelineRun %q failed to finish within %q", pr.Name, pr.PipelineTimeout(ctx).String()),
+	})
+}
```

Comment:
consider defining this as a member function in [pipelinerun_types.go](https://github.com/tektoncd/pipeline/blob/main/pkg/apis/pipeline/v1beta1/pipelinerun_types.go#L159) with tests:
```suggestion
// SetTimeoutCondition sets the status of the PipelineRun to timed out.
func (pr *v1beta1.PipelineRun) SetTimeoutCondition(ctx context.Context) {
	pr.Status.SetCondition(&apis.Condition{
		Type:    apis.ConditionSucceeded,
		Status:  corev1.ConditionFalse,
		Reason:  v1beta1.PipelineRunReasonTimedOut.String(),
		Message: fmt.Sprintf("PipelineRun %q failed to finish within %q", pr.Name, pr.PipelineTimeout(ctx).String()),
	})
}
```